### ISSUE TITLE
Investigate why some instances of the gallery continued to use the broken search service after it was rolled back

### ISSUE THREAD

**Issue by scottbommarito** [NuGet/NuGetGallery#3835]
After we rolled back the search service, some instances of the gallery still failed on OData requests. We should investigate how and why these instances did not fix themselves, but some instances did.

**Comment by scottbommarito** [comment]
@joelverhagen and I did some investigations and we believe we've figured it out.

C#'s `HttpClient` pools connections. In other words, if a request is made to a service that was recently connected to, the previous connection is used. This provides performance improvements.

At the same time, an Azure VIP swap does not terminate connections.

As a result, the gallery was pooling the old search service connection and did not begin using the new search service.

To test this, we used fiddler and were able to successfully verify locally that the gallery pools it's connections to the search service.

This explains a number of things:
- Why all gallery instances appeared to adopt the broken search service progressively over time.
- Why all gallery instances appeared to adopt the rolled back search service progressively over time.
- Why the instances that were still pooling an old connection worked perfectly fine after being rebooted.

In the future, we should be cognizant that it can take some time before the new search service is adopted by the gallery (unless we shut down the staging slot). Fortunately, this breaks our fundamental deployment principal, which is that all deployments must be capable of being rolled back.

Closing as this is solved.
